### PR TITLE
Lunr search

### PR DESCRIPTION
**Pull request by pwright**

### CODE REVIEW

**Comment by pwright** [comment]
This is a simple search, that's totally within our control, and provides a good search experience.

**Comment by pwright** [comment]
@riprasad could you take a look at this PR?

**Comment by EricWittmann** [comment]
Hey @pwright - thanks for this!  Sorry it's taken me so long to have a look at it.  I'll give it a try locally today (if I can remember how to do that). :) :)

**Comment by EricWittmann** [comment]
@pwright When is the `package.json` file used?  It doesn't contain any useful scripts, and my naive understanding of that file is that it's used e.g. when doing `npm install`.  But I don't think we're doing that... I'm sure it's used in other contexts.

**Comment by pwright** [comment]
@EricWittmann  right,  package.json isn't used currently. 
It could be used instead of installing stuff globally, I think that was my plan until I read the DockerFile (where antora is installed globally). Best thing is to remove it?

**Comment by EricWittmann** [comment]
I'm playing with it now.  The docker approach to the build is useful but also problematic.  It might be nice to have a better way to install packages locally and still do the build.  I'll see what I can figure out. :)

**Comment by EricWittmann** [comment]
This looks really nice, thanks @pwright !  I haven't looked at the implementation much - how easy is the UI to tweak?  Here's what I see when I do a search:

image

It would be better if we could make the search results panel a bit wider, and also address the amount of vertical space that each search result item is taking up - either by preventing wrapping or by adjusting the line height.

**Comment by EricWittmann** [comment]
After looking some more, I see that there is some CSS that controls the visuals and is easily changeable.  All good.  :)

If there are no objections from @Apicurio/developers I will make some tweaks to the CSS and then merge this PR.  Thanks @pwright !

**Comment by EricWittmann** [comment]
@pwright Oh by the way - let's keep the `package.json` - it was pretty useful actually to install the requirements locally and then invoke `build.sh` directly.  This wasn't recommended in the past because it wouldn't run correctly on all operating systems (hence the docker image).  With the latest versions of everything, that is no longer a problem.  So the docker image now has limited utility.  Fortunately the `build.sh` can be run directly.  I'll update the README to reflect this I think.

File: supplemental_ui/css/search.css [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+.search-result-highlight {
+  color: #174d8c;
+  background: rgba(143, 187, 237, 0.1);
+  padding: .1em .05em;
+}
+
+.search-result-item {
+  display: flex;
```
Comment (pwright) [inline supplemental_ui/css/search.css:81]: @EricWittmann  I think you wanna set  a height after this line

File: supplemental_ui/css/search.css [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  top: 100%;
+  border-radius: 4px;
+  margin: 6px 0 0;
+  padding: 0;
+  text-align: left;
+  height: auto;
+  background: transparent;
+  border: none;
+  max-width: 600px;
+  min-width: 500px;
```
Comment (pwright) [inline supplemental_ui/css/search.css:48]: @EricWittmann  I think you wanna increase min-width (and max-width too)

**Pull request merged**